runtime/subprocess_windows/src/widestr.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+use std::ffi::OsStr;
+use std::fmt;
+use std::ops::Index;
+use std::ops::Range;
+use std::ops::RangeFrom;
+use std::ops::RangeTo;
+use std::os::windows::ffi::OsStrExt;
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct WCString {
+  buf: Box<[u16]>,
+}
```

Comment:
Why is this needed and not something like `std::ffi::OsString`?

Reply:
`OsString` is WTF-8 encoded on windows, so any time you want to call into a windows api you have to re-encode it to null-terminated utf16. This type just represents a null terminated utf16 string